c10/core/DeviceType.h

```diff
@@ -12,6 +12,23 @@
 
 namespace c10 {
 
+// These contains all device types that also have a BackendComponent
+// and therefore participate in per-backend functionality dispatch keys.
+// This is most backends except PrivateUse2 and PrivateUse3
```

Comment:
I guess we could just add `DeviceType::PrivateUse2` and `DeviceType::PrivateUse3` and call it a day (and remove the need for this extra macro). Probably doesn't matter too much though.

Reply:
There was a bunch of extra boilerplate I needed to do for that (making "privateuse2" parse in device field, e.g.), so I did this macro instead.